apps/api/lib/utils/isValidBase64.ts

```diff
@@ -0,0 +1,17 @@
+export function isValidBase64(input: string): boolean {
+  try {
+    let base64Data = input;
+    // Check if the input starts with a data URI scheme (data:content/type;base64,...)
+    if (input.startsWith("data:")) {
+      const parts = input.split(",");
+      if (parts.length === 2) {
+        // metadata and base64Data
+        base64Data = parts[1]; // Extract the base64 data from the second part
+      }
+    }
+    const buffer = Buffer.from(base64Data, "base64").toString("base64");
```

Comment:
```suggestion

    if (!base64Regex.test(base64Data)) return false;
    
    Buffer.from(base64Data, "base64");
    return true;
```

We can use regex to test and return false if it doesn't adhere to the regex pattern. The next line would fail if not base64, at which point it should return false thanks to catch.

Reply:
I think we no longer need to use this line too anymore, right?
``` Buffer.from(base64Data, "base64"); ```

as the condition before it should be enough to cover base64 validity

Reply:
Actually we still do.

- The regex test checks if the string looks like valid base64
- The buffer conversion checks if the string can be successfully decoded as base64

Reply:
@ThyMinimalDev 
What do you think?
The regex **is** fairly strict so there shouldn't be many strings that pass the regex and fail Base64, but what is the safer approach here?

Reply:
I Think using a  regexp is the best approach here, I don't think there is a bulletproof way to make sure that the base64 string is actually a valid image, but as long as it's a valid base64 it is fine.